src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalyzer.cs

```diff
@@ -23,12 +23,12 @@ public abstract class DiagnosticAnalyzer
 
         public sealed override bool Equals(object obj)
         {
-            return ReferenceEqualityComparer.Instance.Equals(this, obj);
+            return ReferenceEquals(this, obj);
         }
 
         public sealed override int GetHashCode()
         {
-            return ReferenceEqualityComparer.Instance.GetHashCode(this);
+            return ReferenceEqualityComparer.GetHashCode(this);
```

Comment:
`System.Runtime.CompilerServices.RuntimeHelpers.GetHashCode(this);` or, alternately, `base.GetHashCode();`